ruby/neo4j/driver/config.rb

```diff
@@ -42,39 +66,12 @@ def java_config
         fetch(:java_config)
       end
 
-      class << self
-        def default_config
-          {
-            logger: ActiveSupport::Logger.new(STDOUT, level: ::Logger::ERROR), # :set_log
-            leaked_session_logging: false,
-            #connection_liveness_check_timeout: -1, # Not configured
-            max_connection_lifetime: 1.hour, # :set_max_connection_life_time
-            max_connection_pool_size: 100, #:set_max_pool_size
-            connection_acquisition_timeout: 1.minute, #:set_max_connection_acquisition_time
-            encryption: false, # :set_transport
-            connection_timeout: 30.seconds, # BoltSocketOptions_set_connect_timeout
-            idle_time_before_connection_test: -1,
-            max_transaction_retry_time: Internal::Retry::ExponentialBackoffRetryLogic::DEFAULT_MAX_RETRY_TIME,
-            #resolver: nil # :set_address_resolver
-            keep_alive: true, # BoltSocketOptions_set_keep_alive
-            metrics_enabled: false,
-            # ???? BoltConfig_set_user_agent
-            fetch_size: 1000,
-            user_agent: "neo4j-java/#{driver_version}"
-          }
-        end
-
-        def driver_version
-          Neo4j::Driver::VERSION
-        end
-      end
-
       private
 
       def init_security_and_trust_config(config)
-        trust_strategy = TrustStrategy.new(config)
+        trust_strategy = config[:trust_strategy] ? TrustStrategy.new(config) : DEFAULTS[:trust_strategy]
         merge!(
-          security_settings: Neo4j::Driver::Internal::SecuritySetting.new(config[:encrypted], trust_strategy),
+          security_settings: Neo4j::Driver::Internal::SecuritySetting.new(config[:encryption], trust_strategy),
```

Comment:
the internal variable name in java-driver is `encrypted` but we have it as a `encryption` in ruby-driver so for backward compatibility I am keeping it as `encryption`

Reply:
yes, correct the API name in java is `withEncryption` and we derive all config keys from those method names